packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+const apiKeyCredentialPackage = "@azure/core-auth";
+
+function enrichImportedString(
+  sampleGroup: RLCSampleGroup,
+  importedDict: Record<string, Set<string>>,
+  defaultFactoryName: string,
+  packageName: string
+) {
+  const importedTypes: string[] = [];
+  if (!!importedDict[packageName] && importedDict[packageName].size > 0) {
+    const otherTypes = Array.from(importedDict[packageName]).join(", ");
+    importedTypes.push(
+      `import ${defaultFactoryName}, { ${otherTypes} } from "${packageName}";`
+    );
+  } else {
+    importedTypes.push(`import ${defaultFactoryName} from "${packageName}";`);
+  }
+  if (importedDict[tokenCredentialPackage]) {
+    const otherTypes = Array.from(importedDict[tokenCredentialPackage]).join(
+      ", "
+    );
+    importedTypes.push(
+      `import { ${otherTypes} } from "${tokenCredentialPackage}";`
+    );
+  }
+  if (importedDict[apiKeyCredentialPackage]) {
+    const otherTypes = Array.from(importedDict[apiKeyCredentialPackage]).join(
+      ", "
+    );
+    importedTypes.push(
+      `import { ${otherTypes} } from "${apiKeyCredentialPackage}";`
+    );
+  }
+  sampleGroup.importedTypes = importedTypes;
+}
+
+function enrichParameterInSample(
+  sample: RLCSampleDetail,
+  parameters: SampleParameters
+) {
+  sample.clientParamAssignments = getAssignmentStrArray(parameters.client);
+  sample.clientParamNames = getContactParameterNames(parameters.client);
+  sample.pathParamAssignments = getAssignmentStrArray(parameters.path);
+  sample.pathParamNames = getContactParameterNames(parameters.path);
+  sample.methodParamAssignments = getAssignmentStrArray(parameters.method);
+  sample.methodParamNames = parameters.method.length > 0 ? "options" : "";
+}
+
+function getAssignmentStrArray(parameters: SampleParameter[]) {
+  return parameters.filter((p) => !!p.assignment).map((p) => p.assignment!);
+}
+
+function getContactParameterNames(parameters: SampleParameter[]) {
+  return parameters
+    .filter((p) => p.name != null)
+    .map((p) => p.name!)
+    .join(",");
+}
+
+function convertClientLevelParameters(
+  model: RLCModel,
+  importedDict: Record<string, Set<string>>,
+  schemaMap: Map<string, Schema>
+): SampleParameter[] {
+  if (!model.options) {
+    return [];
+  }
+  const clientParams: SampleParameter[] = [];
+  const urlParameters = model?.urlInfo?.urlParameters?.filter(
+    // Do not include parameters with constant values in the signature, these should go in the options bag
+    (p) => p.value === undefined
+  );
+  const { addCredentials, credentialScopes, credentialKeyHeaderName } =
+    model.options;
+  const hasUrlParameter = !!urlParameters,
+    hasCredentials =
+      addCredentials &&
+      ((credentialScopes && credentialScopes.length > 0) ||
+        credentialKeyHeaderName);
+
+  if (hasUrlParameter) {
+    // convert the host parameters in url
+    const clientParamAssignments = urlParameters.map((urlParameter) => {
+      const urlValue = mockParameterTypeValue(
```

Comment:
I would encourage you to also think of the design that if we could get the real values from typespec examples, how can we apply that with minor change of the existing code.

Reply:
updated.